torch/distributed/__init__.py

```diff
@@ -18,6 +29,22 @@ def is_available():
 if is_available() and not torch._C._c10d_init():
     raise RuntimeError("Failed to initialize torch.distributed")
 
+def get_debug_mode():
```

Comment:
Does this method have to be public? Can we keep it private for now and also make the enum `DistributedDebugMode` private? I'm assuming these are only for internal use and users would only be setting `TORCH_DISTRIBUTED_DEBUG`?

Reply:
I thought we want OSS users to use the debug mode features as well? it will be helpful for OSS users to debug issues

Reply:
I think the reasoning is that PyTorch user themselves won't have a need to call this function or use the enum added here, it will only be used in pytorch code, so we should _ it.